srml/system/src/lib.rs

```diff
@@ -14,8 +14,48 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! System manager: Handles lowest level stuff like depositing logs, basic set up and take down of
+//! # System module
+//! 
+//! The system module provides low-level access to core types and cross-cutting utilities.
+//! It acts as the base layer, for other SRML modules, to interact with the Substrate framework components.
+//! To use it in your module, you need to implement the system [`Trait`].
+//! 
+//! ## Overview
+//! 
+//! The system module defines the core data types used in Substrate runtime. 
+//! It also provides several utility functions ([`Module`]) for other modules.
+//! 
+//! In addition, it manages the storage items for extricsics data, indexes, event record and digest items, among other things, 
+//! which support the execution of the current block.
+//! 
+//! It also handles lowest level stuff like depositing logs, basic set up and take down of
 //! temporary storage entries, access to old block hashes.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions
+//! 
+//! The system module does not implement any dispatchable functions.
+//! 
+//! ### Public functions - [`Module`]
+//! 
```

Comment:
I assume this empty sub-section is a forgotten todo?

Reply:
No, it just links to the [`Module`] which has all the public functions documented. Adding them here again was duplicating content.

Reply:
I think then put a single sentence below the heading:
`All public functions are implement by [`Module`]` or something similar.